reference/constraints/When.rst

```diff
@@ -215,11 +223,22 @@ validation based on its value:
 
         class Discount
         {
+            // using an expression
             #[Assert\When(
                 expression: 'value == "percent"',
                 constraints: [new Assert\Callback('doComplexValidation')],
             )]
             private ?string $type;
+
+            // using a closure
+            #[Assert\When(
+                expression: static function (Discount $discount) {
+                    return $discount->getType() === 'percent';
+                },
+                constraints: [new Assert\Callback('doComplexValidation')],
+            )]
+            private ?int $amount;
```

Comment:
It reads weird if the closure and expression are the same, but the property name is different `$type !== $amount` 🤔